test-app/tests/acceptance/routeable-engine-demo-refresh-test.js

```diff
@@ -0,0 +1,48 @@
+import { module, test } from 'qunit';
+import { setupApplicationTest } from 'ember-qunit';
+import { visit, find, click } from '@ember/test-helpers';
+
+module('Acceptance | routeless engine demo', function (hooks) {
+  setupApplicationTest(hooks);
+
+  module('Engine Router Service Refresh Method', function () {
+    test('refresh without params triggers refresh with current route', async function (assert) {
+      await visit('/routable-engine-demo/ember-blog/new');
+
+      let counter = await find('.route-refresh-counter').textContent;
+      assert.dom('.route-refresh-counter').hasText(counter);
```

Comment:
@aklkv this assertions seems strange.
we take text of the element and then assert that element has the text we have just got on previous line.. as assertions have to be status, I assume you wanted here

```suggestion
      assert.dom('.route-refresh-counter').hasText(1);
```

and then, test should be

```
await click('.refresh');
assert.dom('.route-refresh-counter').hasText(2);
```

Reply:
I think I was battling some of the none-determinism when running test along side the other once where the state would accumulate and I wouldn't know what the initial state is, I will look into simplifying this since this test suite is now separated

Reply:
Ok, I have removed useless assertions but I still think we should keep other things else as state persists between refreshes so counter will increment on the background